PowerEditor/src/MISC/Common/FileInterface.cpp

```diff
@@ -0,0 +1,167 @@
+// This file is part of Notepad++ project
+// Copyright (C)2019 Don HO <don.h@free.fr>
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either
+// version 2 of the License, or (at your option) any later version.
+//
+// Note that the GPL places important restrictions on "derived works", yet
+// it does not provide a detailed definition of that term.  To avoid
+// misunderstandings, we consider an application to constitute a
+// "derivative work" for the purpose of this license if it does any of the
+// following:
+// 1. Integrates source code from Notepad++.
+// 2. Integrates/includes/aggregates Notepad++ into a proprietary executable
+//    installer, such as those produced by InstallShield.
+// 3. Links to a library or executes a program that does any of the above.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+
+#include "FileInterface.h"
+
+
+CFile::CFile(const char *fname, Mode fmode) : _hMode(fmode)
+{
+	if (fname)
+	{
+		DWORD access, share, disp, attrib;
+		fillCreateParams(access, share, disp, attrib);
+
+		_hFile = ::CreateFileA(fname, access, share, NULL, disp, attrib, NULL);
+	}
+	else
+	{
+		_hFile = INVALID_HANDLE_VALUE;
+	}
+
+	if ((_hFile != INVALID_HANDLE_VALUE) && (_hMode == APPEND))
+	{
+		LARGE_INTEGER offset;
+		offset.QuadPart = 0;
+
+		::SetFilePointerEx(_hFile, offset, NULL, FILE_END);

```

Comment:
See https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-setfilepointerex

> You can use SetFilePointerEx to determine the length of a file. To do this, use FILE_END for dwMoveMethod and seek to location zero. The file offset returned is the length of the file. However, this practice can have unintended side effects, such as failure to save the current file pointer so that the program can return to that location. It is simpler and safer to use the GetFileSizeEx function instead.